cpp/memilio/io/README.md

```diff
@@ -1,92 +1,95 @@
+This directory contains utilities for reading and writing data from and to files in different formats. The main part is a serialization framework that can be used to define the structure of data without using a specific file format. There are implementations of the framework for different formats. The framework is described in detail below, also see the [serialization example](../../examples/serialize.cpp).
 
 ## The Serialization framework
 
-Main functions and types:
--------------------------
-- functions serialize and deserialize: 
+## Main functions and types
+
+- functions serialize and deserialize:
      Main entry points to the framework to write and read values, respectively. The functions expect an IOContext
      (see Concepts below) that stores the serialized data. (De-)serialization can be customized by providing a
-     (de-)serialize_internal overload or a (de-)serialize member function for the type. See the section "Implementing 
+     (de-)serialize_internal overload or a (de-)serialize member function for the type. See the section "Implementing
      serialization for a new type" or the documentation for `serialize` and `deserialize`.
-- IOStatus and IOResult: 
+- IOStatus and IOResult:
      Used for error handling, see section "Error Handling" below.
 
-Concepts: 
----------
+## Concepts
+
 1. IOContext
 Stores data that describes serialized objects of any type in some unspecified format and provides structured
 access to the data for deserialization. Implementations of this concept may store the data in any format
 they want including binary. The data may also be written directly to disk. The context also keeps track
 of errors. An IOContext object `io` allows the following operations:
-- `io.create_object("Type")`: 
-     Returns an IOObject for the type called `"Type"`. The IOObject (see below) allows adding data that describes 
-     the object to be serialized. The function must return something that can be assigned to a local
-     variable, e.g., a temporary or copyable function. IOObject may store references to the context internally,
-     so the lifetime of the local IOObject may not exceed the lifetime of the IOContext that created it.
-- `io.expect_object("Type")`: 
-     Returns an IOObject for the type called `"Type"`.
-     The IOObject (see below) provides access to the data needed for deserialization. 
-- `io.flags()`:
-     Returns the flags that determine the behavior of serialization; see IOFlags.
-- `io.error()`:
-     Returns an IOStatus object to check if there were any errors during serialization.
-     Usually it is not necessary to check this manually but can be used to report the error faster and 
-     avoid expensive operations that would be wasted anyway.
-- `io.set_error(s)` with some IOStatus object: 
-     Stores an error that was generated outside of the IOContext, e.g., if a value that was deserialized 
-     is outside an allowed range.
-2. IOObject
+
+     - `io.create_object("Type")`:
```

Comment:
Why do you use 5 spaces for indentation instead of 4? Do we use different formatting for Markdown files?

Reply:
lol good catch, i accidently changed it to 5....